ros/angel_system_nodes/angel_system_nodes/task_monitor.py

```diff
@@ -75,26 +76,20 @@ def __init__(self):
                             ' place the coffee filter into the dripper,'
                             ' and then place the dripper on top of the mug')
 
-        # TODO: make a labels.txt file for these steps
-        self.steps = [
-            {'name': 'Pour_12oz_of_water_into_a_liquid_measuring_cup'},
-            {'name': 'Pour_water_from_the_liquid_measuring_cup_into_the_electric_kettle'},
-            {'name': 'Turn_on_the_kettle'},
-            {'name': 'Turn_on_kitchen_scale'},
-            {'name': 'Place_bowl_on_scale'},
-            {'name': 'Zero_scale'},
-            {'name': 'Add_coffee_beans_until_scale_reads_25_grams'},
-            {'name': 'Pour_coffee_beans_into_coffee_grinder'},
-            {'name': 'Take_the_coffee_filter_and_fold_it_in_half_to_create_a_semi_circle'},
-            {'name': 'Fold_the_filter_in_half_again_to_create_a_quarter_circle'},
-            {'name': 'Place_the_folded_filter_into_the_dripper_such_that_the_point' +
-                     '_of_the_quarter_circle_rests_in_the_center_of_the_dripper'},
-            {'name': 'Spread_the_filter_open_to_create_a_cone_inside_the_dripper'},
-            {'name': 'Place_the_dripper_on_top_of_the_mug'},
-            {'name': 'Done'}
-        ]
+        # Load the task steps from the provided steps file
+        with open(task_steps_file, "r") as f:
+            self._task_steps= json.load(f)
+
+        # Create the list of steps
+        self.steps = []
+        for key, value in self._task_steps.items():
+            self.steps.append({'name': key.replace(' ', '_')})
+
+        # Manually add the finish step
+        self.steps.append({'name': 'Done'})
 
         # Create the transitions between steps, assuming linear steps
+        # TODO: use the step index to decide the destination step
```

Comment:
It looks like this is what is happening now, so is this TODO accomplished?

Reply:
Hmm I should have been more clear in the TODO. I meant for this to say that we should use the step indices in the provided task steps config file to decide the `dest` step. Right now it's just still assuming all of the steps are in linear order.